main.tf

```diff
@@ -0,0 +1,138 @@
+/**
+ * Creates a Lambda function with associated role and policies that
+ * will run the packer-janitor tool to get rid of orphaned Packer
+ * instances and their associated resources.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies to describe and delete instances, keypairs,
+ *   and security groups, as well as write messages to Cloudwatch Logs
+ * * Cloudwatch Logs group
+ * * Cloudwatch Event to regularly run cleanup job
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "packer-janitor-lambda" {
+ *   source                 = "trussworks/lambda-packerjanitor/aws"
+ *   packer_resource_delete = "true"
+ *   packer_timelimit       = "4"
+ *   job_identifier         = "sample_app"
+ *   s3_bucket              = "sample_app_lambdas"
+ *   version_to_deploy      = "2.8"
+ * }
+ * ```
+ *
+ * For more details on the capabilities of the packer-janitor tool, as
+ * well as how to deploy it, see <https://github.com/trussworks/truss-aws-tools>.
+ */
+
+locals {
+  pkg  = "truss-aws-tools"
+  name = "packer-janitor"
+}
+
+data "aws_region" "current" {}
+
+# This is the main policy this job is going to need.
+data "aws_iam_policy_document" "main" {
+  # Allow describing instances, keypairs, and security groups.
+  statement {
+    sid    = "Describes"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeSecurityGroups",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow terminating EC2 Instances
+  statement {
+    sid    = "TerminateInstances"
+    effect = "Allow"
+
+    actions = [
+      "ec2:TerminateInstances",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow deleting key pairs.
+  statement {
+    sid    = "DeleteKeyPair"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DeleteKeyPair",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow deleting security groups.
+  statement {
+    sid    = "DeleteSecurityGroup"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DeleteSecurityGroup",
+    ]
+
+    resources = ["*"]
```

Comment:
Similarly, do SGs get tagged by Packer? Wondering if we can set a condition such that the lambda can only delete those SGs.

Reply:
Unfortunately, security groups don't get tagged by packer yet. They all have the *name* "packer_somethingsomething", but no tag in the current version.

Reply:
Oh, maybe you can use the Name tag here and use a splat to match (`packer_*`)? I'm not totally sure that's an option, but I think it's worth looking into. Restricting what SGs can get deleted would be a Good Thing™.

Reply:
Unfortunately, it's not the Name *tag*, it's the "name" of the SG, which is a different field, so I don't think that will work. I agree it would be good to restrict that, but at the moment I don't think it can be. :/